### DISCUSSION TITLE
Status 403 when deleting messages

### DISCUSSION THREAD

**Discussion by RomanticFalconXL** [victornpb/undiscord#483] [1. Questions & Answers]
Hey guys! This is my first post here and I need your help.

I've been using Undiscord to mass-delete my own messages in a Discord server. The server uses DynoBot, so when I run the tool, it detects DynoBot's messages as my own messages and tries to delete them but fails to do so, thus giving error 403 messages. It runs through the same set of DynoBot messages during every single page, and it's causing the script to run slowly and eventually crash the browser tab.

Any ideas on how to stop Undiscord from trying to delete DynoBot messages?

**Comment by Adilius** [comment]
Ran into same issue, this is what I did.

- Running the script in other channels that does not allow bot commands (so anything except for #bot_commands or alike). You can supply several channel IDs by seperating them with comma and then excluding bot command channel; however I found no easy way to extract all channel IDs from a server.
- Adding `.` to the regex filter. This makes the bot skip bot commands with no reponse text or text in embeds (which are most of them)

However I am waiting for an update to fix trying to delete messages written by bots as a response to bot commands usage when you lack the permissions to delete those messages.

**Reply by VoidPr1me** [reply]
Great hint!
Just one question: how does one add . to the regex filter? Can you please explain step-by-step as I would like to do that.

**Comment by 573rn** [comment]
If the 403 error is because of a bot:

In Regex enter this:

`^(?!\s*$)(?!.*\bwarn\b).*$`

This works specifically with a GPT bot on the server I'm in.

(?!\s*$) = Whitespace: the bot in my server often posts posts that are only whitespace, aka: no characters at all. This skips posts with no input.

(?!.*\bwarn\b) = The other post the bot makes contains the word "Warn", this skips all posts that contain the word "Warn"

Replace these with whatever unique words or phrases the bot in the server you are trying to avoid uses.

**Reply by ghost** [reply]
Where do I add it?
